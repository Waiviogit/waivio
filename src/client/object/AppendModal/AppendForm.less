@import (reference) '../../styles/custom.less';
@import (reference) '../../styles/modules/variables.less';

.AppendForm {
  padding-bottom: 24px;

  .ant-select {
    margin: 0 0 10px;
  }

  .ant-form-item {
    margin-bottom: 0;
    padding-bottom: 24px;
  }

  &__hidden {
    display: none !important;
  }

  &__previewWrap {
    display: flex;
    justify-content: center;
    background-size: cover;

    .avatar {
      background-repeat: no-repeat;
      background-position: 50% 50%;
      border: @base-border;
    }

    .background {
      display: flex;
      justify-content: space-between;
      align-items: center;
      background-color: @white;
      border: 1px solid @blue-hawkesBlue;
      border-radius: 4px;
      margin-top: 5px;
      padding: 15px;
      background-size: cover;
      background-position: center;
      width: 100%;
      height: 200px;

      @media @xsmall {
        height: 67px;
      }
    }
  }

  .NewLineButton {
    display: flex;
    align-items: center;
    cursor: pointer;

    .anticon-plus-circle {
      color: @primary-color;
      font-size: 23px;
      margin: 10px 10px 10px 0;
    }

    &:hover {
      color: @primary-color;
    }
  }

  .NewsFiltersRule-title {
    margin-top: 5px;
  }

  .NewsFiltersRule-line {
    .ObjectCard {
      padding: 5px 0;
      border-bottom: none;
    }

    &-and {
      padding: 17px 10px;
    }

    &-card {
      position: relative;
      padding-left: 5px;
      border: @base-border;
      border-radius: 4px;
      margin-bottom: 10px;
    }

    &-close {
      position: absolute;
      top: -12px;
      right: -12px;
      background-color: @white;

      .anticon {
        font-size: 23px;
        cursor: pointer;

        &:hover {
          color: @primary-color;
        }
      }
    }

    &-search {
      padding: 10px 0;
    }

    padding: 10px 0;
    border-bottom: @base-border;
  }

  & .ant-form-item-label {
    position: relative;
    top: 8px !important;
  }

  & .ant-form-item-required {
    &:before {
      content: '' !important;
    }
  }

  & .ant-checkbox-wrapper > :last-child {
    vertical-align: -3px;
  }

  &__label {
    font-weight: bold;
  }

  &__appendTitles {
    font-weight: bold;
    color: @tertiary-color;
    margin-left: 5px !important;
  }

  &__input {
    height: 45px !important;
    font-size: 20px !important;
    font-weight: bold;
    background-color: unset !important;
    width: 100%;
    margin: 0 0 10px !important;

    &.validation-error {
      border: 1px solid @red;
      box-shadow: 0 0 3px @orange, 0 0 5px @red;
    }
  }

  &__hashtags {
    & > div {
      padding-top: 5px !important;
      padding-bottom: 8px !important;
      height: auto !important;
    }
  }

  &__bottom {
    display: flex;
    justify-content: space-between;
    align-items: center;

    &__info {
      font-weight: bold;

      .iconfont {
        font-size: 20px;
        vertical-align: -3px;
        margin-right: 4px;
      }
    }

    &__right {
      display: flex;
      align-items: center;

      &__saving {
        margin-right: 16px;
      }
    }

    &__submit {
      margin-top: 10px !important;
      margin-bottom: 0 !important;

      .ant-form-item-control {
        text-align: right !important;
      }
    }

    &__cancel {
      margin-bottom: 0 !important;
      padding-right: 10px;
      margin-right: 10px !important;

      .ant-form-item-control {
        text-align: right !important;
      }

      .Action {
        background-color: @red !important;
      }
    }
  }

  .ant-select-selection__choice {
    font-weight: 500 !important;
    color: @topic-button-color !important;
    background-color: @topic-button-background-color !important;
    border-radius: @border-radius-base !important;
    border: @topic-button-border;

    & > div {
      position: relative;
      top: -2px;
      font-size: 14px;
    }

    & > span {
      color: @black !important;
    }
  }

  .CreateObject__row {
    margin-top: -5px;
  }

  &__drop-down.ant-select-dropdown.ant-select-dropdown--single.ant-select-dropdown-placement-bottomLeft {
    z-index: 1300;
  }
}

div[data-dir='rtl'] .Editor {
  .ant-form-item-label {
    text-align: right;
  }

  &__reading_time {
    float: left;
  }
}

.append-combined-value__validation-msg {
  color: @red-danger;
}

.image-wrapper {
  display: flex;
  flex-direction: column;

  .QuickPostEditor__footer {
    padding-left: 0;
  }

  & > span {
    padding-left: 4px;
    margin-bottom: 16px;
  }
}

@import './AppendForm-nightmode';
