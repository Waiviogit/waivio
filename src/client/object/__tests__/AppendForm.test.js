import React from 'react';
import { shallow } from 'enzyme';
import { IntlProvider } from 'react-intl';
import AppendForm from '../AppendModal/AppendForm';

describe('(Component) AppendForm', () => {
  it('renders and matches snapshot', () => {
    const props = {
      defaultVotePercent: 20,
    };
    const wrapper = shallow(
      <IntlProvider>
        <AppendForm {...props} />
      </IntlProvider>,
    );
    expect(wrapper).toMatchSnapshot();
  });
});
