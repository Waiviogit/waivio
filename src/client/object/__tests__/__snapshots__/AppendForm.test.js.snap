// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`(Component) AppendForm renders and matches snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <IntlProvider>
    <UNDEFINED
      defaultVotePercent={20}
    />
  </IntlProvider>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "defaultVotePercent": 20,
    },
    "ref": null,
    "rendered": null,
    "type": Object {
      "$$typeof": Symbol(react.memo),
      "WrappedComponent": [Function],
      "compare": null,
      "displayName": "Connect(Form(Connect(InjectIntl(withEditor(InjectIntl(AppendForm))))))",
      "type": [Function],
      Symbol(Symbol.hasInstance): [Function],
    },
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "defaultVotePercent": 20,
      },
      "ref": null,
      "rendered": null,
      "type": Object {
        "$$typeof": Symbol(react.memo),
        "WrappedComponent": [Function],
        "compare": null,
        "displayName": "Connect(Form(Connect(InjectIntl(withEditor(InjectIntl(AppendForm))))))",
        "type": [Function],
        Symbol(Symbol.hasInstance): [Function],
      },
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
  Symbol(enzyme.__childContext__): Object {
    "intl": Object {
      "defaultFormats": Object {},
      "defaultLocale": "en",
      "formatDate": [Function],
      "formatHTMLMessage": [Function],
      "formatMessage": [Function],
      "formatNumber": [Function],
      "formatPlural": [Function],
      "formatRelative": [Function],
      "formatTime": [Function],
      "formats": Object {},
      "formatters": Object {
        "getDateTimeFormat": [Function],
        "getMessageFormat": [Function],
        "getNumberFormat": [Function],
        "getPluralFormat": [Function],
        "getRelativeFormat": [Function],
      },
      "locale": "en",
      "messages": Object {},
      "now": [Function],
      "onError": [Function],
      "textComponent": "span",
      "timeZone": null,
    },
  },
}
`;
